lua/entities/emplacements_turret_base/shared.lua

```diff
@@ -57,8 +70,23 @@ function ENT:ShooterStillValid()
     elseif CLIENT then
         shooter = self:GetDTEntity( 0 )
     end
+    return IsValid( shooter ) and shooter:Alive() and ( ( self:GetPos() + self.TurretModelOffset ):Distance( shooter:GetShootPos() ) <= self.emplacementDisconnectRange )
```

Comment:
Splits the logic out a bit and uses a more efficient distance method.

You could save the `maxDistanceSqr` variable wherever you set `emplacementDisconnectRange` too if you wanted.

```suggestion
    local shooterIsValid = IsValid( shooter ) and shooter:Alive()
    if not shooterIsValid then return false end

    local originPos = self:GetPos() + self.TurretModelOffset
    local maxDistanceSqr = self.emplacementDisconnectRange * self.emplacementDisconnectRange
    local distanceSqr = originPos:DistToSqr( shooter:GetShootPos() )
    
    return distanceSqr <= maxDistanceSqr
```